### DISCUSSION TITLE
Cover some of the checks done by flake8 bugbear extension

### DISCUSSION THREAD

**Discussion by mehdigmira** [microsoft/pyright#4725] [Ideas]
Hello,

The scope pyright currently covers is larger than what a static type checker as mypy does, which is great. Stuff such as missing super call, etc. are covered while not really type related.
I think there a few items you could pick up from flake8 and the flake8 bugbear extension (https://pypi.org/project/flake8-bugbear/). 
The ones i find interesting:
- Catching bare excepts: Do not use bare except:, it also catches unexpected events like memory errors, interrupts, system exit, and so on. Prefer except Exception:. If you’re sure what you’re doing, be explicit and write except BaseException:.
- Do not use mutable data structures for argument defaults. They are created during function definition time. All calls to the function reuse this one instance of that data structure, persisting changes between them.

**Comment by erictraut** [comment]
Pyright is a type checker, not a linter. It is intended to complement a general linter like flake8 or pylint. In general, we consider adding diagnostics checks only if they require in-depth knowledge of static types — which are not available to a more general linter. 

Catching bare exceptions is something that a regular linter can easily do.

Pyright already provides a check for mutable types used for argument defaults: `reportCallInDefaultInitializer`. The name is a bit confusing because it started as a check for call expressions, but it was expanded to also cover mutable types like dicts, lists, sets, etc.

**Reply by mehdigmira** [reply]
Thank you for your answer 🙏
